src/components/pod/pod.test.js

```diff
@@ -13,22 +59,236 @@ const PodWithSoftDelete = () => (
+    it.each(specialCharacters)(
+      "should check className as %s for Pod component",
+      (className) => {
+        CypressMountWithProviders(<PodComponent className={className} />);
+        podComponent().should("have.class", className);
+      }
+    );
 
-      podBlock().should("have.css", "background-color", blockBackgroundColor);
+    it.each([
+      ["none", 88, 45],
+      ["extra-small", 120, 61],
+      ["small", 120, 61],
+      ["medium", 136, 77],
+      ["large", 184, 93],
+      ["extra-large", 216, 125],
+    ])("should check %s size for Pod component", (size, height, width) => {
+      CypressMountWithProviders(<PodComponent size={size} />);
+      divPodComponent().then(($el) => {
+        useJQueryCssValueAndAssert($el, "height", height);
+        useJQueryCssValueAndAssert($el, "width", width);
+      });
     });
 
-    it("renders text of children with correct colours", () => {
-      const childrenColor = "rgba(0, 0, 0, 0.65)";
+    it.each([
+      ["primary", "rgb(255, 255, 255)", "none"],
+      ["secondary", "rgb(242, 245, 246)", "none"],
+      ["tertiary", "rgb(237, 241, 242)", "none"],
+      ["tile", "rgb(255, 255, 255)", "rgba(2, 18, 36, 0.2) 0px 2px 3px 0px"],
+      ["transparent", "rgba(0, 0, 0, 0)", "none"],
+    ])(
+      "should check for %s variant pod, the color is %s and boxShadow is %s",
+      (variant, color, boxShadow) => {
+        CypressMountWithProviders(<PodComponent variant={variant} />);
+        divPodComponent()
+          .should(
+            "have.css",
+            "back-ground",
+            `${color} none repeat scroll 0% 0% / auto padding-box border-box`
+          )
+          .and("have.css", "box-shadow", boxShadow);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check title as %s for Pod component",
+      (title) => {
+        CypressMountWithProviders(<PodComponent title={title} />);
+        podTitle().should("have.text", title);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check subtitle as %s for Pod component",
+      (subtitle) => {
+        CypressMountWithProviders(<PodComponent subtitle={subtitle} />);
+        podSubTitle().should("have.text", subtitle);
+      }
+    );
+
+    it.each(["left", "center", "right"])(
+      "should check when title alignment is %s for Pod component",
+      (alignTitle) => {
+        CypressMountWithProviders(<PodComponent alignTitle={alignTitle} />);
+        podTitle().should("have.css", "text-align", alignTitle);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check footer as %s for Pod component",
+      (footerText) => {
+        CypressMountWithProviders(<PodComponent footer={footerText} />);
+        podFooter().should("have.text", footerText).and("be.visible");
+      }
+    );
+
+    it.each([
+      [true, "be.visible"],
+      [false, "not.exist"],
+    ])(
+      "should check when softDelete is %s for Pod component",
```

Comment:
```suggestion
      "should check visibility of undo button when softDelete is %s",
```
Personally I think we should make it clear what this test is actually verifying here. This makes it easier to understand when it is being run.